domain/consensus/model/pow/pow.go

```diff
@@ -0,0 +1,55 @@
+package pow
+
+import (
+	"github.com/kaspanet/kaspad/domain/consensus/model/externalapi"
+	"github.com/kaspanet/kaspad/domain/consensus/utils/consensushashing"
+	"github.com/kaspanet/kaspad/domain/consensus/utils/hashes"
+	"github.com/kaspanet/kaspad/domain/consensus/utils/serialization"
+	"github.com/kaspanet/kaspad/util"
+	"github.com/pkg/errors"
+	"math/big"
+)
+
+// CheckProofOfWork check's if the block has a valid PoW according to its Bits field
+// it does not check if the difficulty itself is valid or less than the maximum for the appropriate network
+func CheckProofOfWork(header *externalapi.DomainBlockHeader, optionalTarget *big.Int) bool {
+	if optionalTarget == nil {
+		optionalTarget = util.CompactToBig(header.Bits)
+	}
+	// The block hash must be less than the claimed target unless the flag
+	// to avoid proof of work checks is set.
+	powNum := hashes.ToBig(calcPowValue(header))
```

Comment:
I'd expect clacPowValue to do the hashes.ToBig().
It's pretty weird to the outside onlooker that calcPowValue returns *externalapi.DomainHash

Reply:
You're right. I did that because it's internal and I didn't care much, but I'll change it